freecad 1.1.0, vtk9.6.0, boost1.88, hdf5 1.14.6, pcl 1.15, occt7.9.3, no cdt's, qt6.11, pyside6 611

looooo opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

regro-cf-autotick-bot committed:
Rebuild for pcl 1.15.0

regro-cf-autotick-bot committed:
Rebuild for hdf5 1.14.6

regro-cf-autotick-bot committed:
Rebuild for libboost 1.88

looooo committed:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.22.18.58.01

looooo committed:
update to version 1.0.2

looooo committed:
reset build number

looooo committed:
occt790

looooo committed:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.22.18.58.01

looooo committed:
migrate to vtk942

looooo committed:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.22.18.58.01

looooo committed:
delete hdf51144 migration

looooo committed:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.22.18.58.01

h-vetinari commented:
You're missing a [dependency](https://github.com/conda-forge/gmsh-feedstock/pull/105) for adding the occt [migration](https://conda-forge.org/status/migration/?name=occt790) AFAICT


<img width="1062" height="556" alt="image" src="https://github.com/user-attachments/assets/5434fc00-ec4c-4079-8e10-505c226611de" />

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ Use of `cdt(mesa-libgl-devel)` is deprecated. Use one or more of `libgl-devel`, `libegl-devel`, `libgles-devel`, `libglx-devel`, and `libopengl-devel`. See https://conda-forge.org/docs/maintainer/knowledge_base/#libgl



<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/23810035408](https://github.com/conda-forge/conda-forge-webservices/actions/runs/23810035408). Examine the logs at this URL for more detail.</sub>

h-vetinari committed:
MNT: Re-rendered with conda-smithy 3.53.1 and conda-forge-pinning 2025.11.19.18.25.31

Other tools:
- conda-build 25.9.0
- rattler-build 0.49.0
- rattler-build-conda-compat 1.4.9

h-vetinari committed:
help solver avoid getting lost

looooo committed:
add migrations for vtk, pcl, python

Commit:
MNT: Re-rendered with conda-smithy 3.53.1 and conda-forge-pinning 2025.11.19.18.25.3

h-vetinari commented:
Sigh, https://github.com/conda-forge/occt-feedstock/pull/130 only landed for occt 7.9.1, so we also need at least https://github.com/conda-forge/gmsh-feedstock/pull/106

looooo committed:
move to occt791

looooo committed:
MNT: Re-rendered with conda-smithy 3.53.2 and conda-forge-pinning 2025.11.21.15.16.01

looooo committed:
disable py313 patch

h-vetinari commented:
The osx error is likely "fixable" (for now) by staying on clang 18. On linux, we have a bigger problem

```
$SRC_DIR/src/Mod/ReverseEngineering/App/SurfaceTriangulation.cpp:43:10: fatal error: pcl/point_traits.h: No such file or directory
   43 | #include <pcl/point_traits.h>
      |          ^~~~~~~~~~~~~~~~~~~~
```

in that [pcl 1.15](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Fpcl-1.15.1-h5abca63_2.conda&with_broken=true) has removed the `point_traits.h` header, and so this needs to be adapted in the source (or we need to rebuild [pcl 1.14](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Fpcl-1.14.1-h679aaff_7.conda&with_broken=true) for all the dependency updates being proposed in this PR)

looooo committed:
use latest available source

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20691236656) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20691236656](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20691236656). Examine the logs at this URL for more detail.</sub>

looooo committed:
version

looooo committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2025.12.30.19.55.26

looooo committed:
remove pivy from host dependencies

looooo committed:
do not check for pivy during build

looooo committed:
add lark as build dependency

looooo committed:
some fixes

looooo commented:
@conda-forge/core

the osx-build is failing with:
```
 ImportError: dlopen($PREFIX/lib/FreeCADGui.so, 0x0002): Symbol not found: __ZTINSt3__117bad_function_callE
  Referenced from: <041F34D1-448B-3EFD-BD5E-C49C7474279B> $PREFIX/lib/libFreeCADGui.dylib
  Expected in:     <994C93A6-44F3-3281-ABEA-6B8E97918A5F> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
``
any ideas?

looooo committed:
try to fix the windows builds

looooo committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2026.01.28.03.26.15

looooo committed:
don't stop build due to failing tests

looooo committed:
Update MACOSX_SDK_VERSION to 15.5

looooo commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2026.01.29.11.55.1

looooo committed:
try to fix windows builds

looooo committed:
try another workaround for windows

looooo committed:
try to fix windows

oursland commented:
@looooo Unfortunately, we have not upgraded FreeCAD to OCCT 7.9.x.  The current pinning is for `occt>=7.8,<7.9`.  

@pieterhijma is the OCCT liason.  Pieter, do you know what outstanding issues prevent the migration to OCCT 7.9 or newer?

looooo committed:
use version 1.1rc2

looooo committed:
Update FreeCAD version to 1.1.0 in meta.yaml

looooo committed:
remove cdt's

looooo committed:
change to v1 recipe format

looooo committed:
remove qt5 things

looooo committed:
MNT: Re-rendered with conda-smithy 3.58.0 and conda-forge-pinning 2026.03.26.13.31.28

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

looooo committed:
Merge branch 'main' into v102

looooo committed:
MNT: Re-rendered with conda-smithy 3.58.0 and conda-forge-pinning 2026.03.26.13.31.28

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

looooo committed:
build native

looooo committed:
MNT: Re-rendered with conda-smithy 3.58.0 and conda-forge-pinning 2026.03.26.13.31.28

looooo committed:
switch back to cross compiling

looooo committed:
MNT: Re-rendered with conda-smithy 3.58.0 and conda-forge-pinning 2026.03.26.13.31.28

looooo commented:
@jschueller any ideas what causes this error:
```
  │ │ CMake Error at $PREFIX/lib/cmake/PySide6/PySide6Config.cpython-312-aarch64-linux-gnu.cmake:15 (message):
 │ │   File or directory
 │ │   $PREFIX/typesystems
 │ │   referenced by variable PYSIDE_TYPESYSTEMS does not exist !
 │ │ Call Stack (most recent call first):
 │ │   $PREFIX/lib/cmake/PySide6/PySide6Config.cpython-312-aarch64-linux-gnu.cmake:43 (set_and_check)
 │ │ -- Configuring incomplete, errors occurred!
 │ │   $PREFIX/lib/cmake/PySide6/PySide6Config.cmake:5 (include)
 │ │   cMake/FindPySide6.cmake:6 (find_package)
 │ │   cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake:60 (find_package)
 │ │   CMakeLists.txt:113 (SetupShibokenAndPyside)
```

qt6.10 and pyside6 6.10 are used.

looooo committed:
try to fix pyside6 problem with typesystems directory

looooo committed:
set typesystems dir with cmake

looooo committed:
replace paths in pyside6 configuration files

jschueller commented:
> @jschueller any ideas what causes this error:
> 
> ```
>   │ │ CMake Error at $PREFIX/lib/cmake/PySide6/PySide6Config.cpython-312-aarch64-linux-gnu.cmake:15 (message):

it definitely looks like a pyside bug in sources/pyside6/libpyside/CMakeLists.txt:
```
# Install-tree / wheel configuration
set(PYSIDE_PYTHONPATH "")
set(PYSIDE_TYPESYSTEMS "typesystems")
set(PYSIDE_GLUE "glue")
set(PYSIDE_SOVERSION "${pyside6_library_so_version}")

configure_package_config_file(
    "${CMAKE_CURRENT_SOURCE_DIR}/PySide6Config-spec.cmake.in"
    "${CMAKE_CURRENT_BINARY_DIR}/install/PySide6Config${SHIBOKEN_PYTHON_CONFIG_SUFFIX}.cmake"
    INSTALL_DESTINATION "${LIB_INSTALL_DIR}/cmake/PySide6"
    PATH_VARS PYSIDE_PYTHONPATH PYSIDE_TYPESYSTEMS PYSIDE_GLUE
)
```
they set PYSIDE_TYPESYSTEMS as typesystems in PySide6Config  but it should be PREFIX/share/PySide6/typesystems 

I  have pushed updates in the pyside recipe

https://qt-project.atlassian.net/browse/PYSIDE-3295

looooo committed:
Pin qt6_main to 611

looooo commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-smithy 3.58.1 and conda-forge-pinning 2026.03.26.13.31.2

oursland commented:
@looooo This combination of dependencies may yield a less than desirable user experience.  Currently, FreeCAD is pinned to [VFX Platform](https://vfxplatform.com/), to maintain compatibility with Blender (5.0) and other applications:

* Python 3.11
* Qt 6.8
* Boost 1.86

In addition there are serious regressions with OCCT 7.9.x.

looooo committed:
use qt6_main 6.10

looooo committed:
MNT: Re-rendered with conda-smithy 3.58.1 and conda-forge-pinning 2026.03.26.13.31.28

looooo committed:
MNT: Re-rendered with conda-smithy 3.58.1 and conda-forge-pinning 2026.03.26.13.31.28

Other tools:
- conda-build 26.3.0
- rattler-build 0.61.4
- rattler-build-conda-compat 1.4.11

looooo committed:
Add conda_install_tool: pixi to conda-forge.yml

Made-with: Cursor

looooo commented:
> @looooo This combination of dependencies may yield a less than desirable user experience. Currently, FreeCAD is pinned to [VFX Platform](https://vfxplatform.com/), to maintain compatibility with Blender (5.0) and other applications:
> 
>     * Python 3.11
> 
>     * Qt 6.8
> 
>     * Boost 1.86
> 
> 
> In addition there are serious regressions with OCCT 7.9.x.

I understand the issues, but I would like to keep the conda-forge package updated with the pinnings from conda-forge.

arch linux is also  at qt6.11 / boost 1.90 / occt 7.9.3 ...

looooo committed:
Add Boost 1.84 CoinPtr/intrusive_ptr ADL patch for FreeCAD 1.1.0

Declare intrusive_ptr_add_ref/release for SoBase* in ViewProvider and
wire recipe to apply patches/boost184_coin_intrusive_ptr.patch for all
platforms (fixes build with Boost >= 1.84).

Made-with: Cursor

looooo committed:
update patch

looooo committed:
boost184 patch: drop explicit SoBase include

Coin3D exposes SoBase via Inventor/misc/SoBase.h; FreeCAD already pulls
the needed definitions through other Inventor headers. Remove the extra
include line to avoid wrong paths and fix the cpp hunk line numbers.

Made-with: Cursor

looooo commented:
@oursland what should we do with this PR and maybe more interesting, what standards should we follow for the dependencies? In my mind we should use the conda-forge pinnings for the conda-forge package and promote this package more as a rolling release package. With conda/pixi it's also always possible to install older deps, so you can choose between stability and actuality... (at least in theory).
maybe you can briefly share your thought on this topic.

adrianinsaval commented:
at the end of the day conda-forge package is a third party package not official freecad release, it should follow what conda-forge is using and cater to conda-forge users, not to our whims. Blender compat I believe comes down to python version only if I'm not mistaken, conda-forge does builds for many python versions so that should be covered (correct me if I'm wrong) but even if that weren't the case the official freecad release provides that so it should not be a concern for conda-forge package, in any case maybe we can provide a separate variant/label that has a specific set of dependencies

looooo committed:
add migration for qt611

looooo committed:
MNT: Re-rendered with conda-smithy 3.61.1 and conda-forge-pinning 2026.04.21.11.53.26

jschueller commented:
I think you need to manually pin vtk to 9.6.1 for qt 6.11 as 9.6.0 is only available for for 6.10

h-vetinari commented:
I would suggest to stop increasing pins; you had at least one passing run with a combination of dependencies that would be good enough to publish. Then individual migrations can be handled by migrators going forward

h-vetinari commented:
> arch linux is also at qt6.11 / boost 1.90 / occt 7.9.3 ...

We haven't started the migration for boost 1.90 yet (though I'm planning to do that ~soon).

looooo committed:
vtk961

looooo committed:
MNT: Re-rendered with conda-smithy 3.61.1 and conda-forge-pinning 2026.04.23.11.42.25

h-vetinari commented on recipe/conda_build_config.yaml:
```diff
@@ -2,6 +2,6 @@ c_compiler:           # [win]
   - vs2022            # [win]
 cxx_compiler:         # [win]
   - vs2022            # [win]
```
This can be dropped BTW, it's the default in conda-forge since a while

looooo committed:
Revert "MNT: Re-rendered with conda-smithy 3.61.1 and conda-forge-pinning 2026.04.23.11.42.25"

This reverts commit 5369bd564efc7e7ee85b690cef6cb8bb4afe3cd4.

looooo committed:
Revert "vtk961"

This reverts commit 3687e30d42d5101a2042686730100eca21f1d817.

looooo committed:
Revert "add migration for qt611"

This reverts commit 5d389dc8c2b9f94b8cc49e38dc2e51697f8b4ed4.

looooo committed:
Revert "MNT: Re-rendered with conda-smithy 3.61.1 and conda-forge-pinning 2026.04.21.11.53.26"

This reverts commit ffbadc3e8e125caed9470a8f1bfe6975c7174a34.

looooo committed:
remove vs2022 specifiers, now default

looooo committed:
MNT: Re-rendered with conda-smithy 3.61.2 and conda-forge-pinning 2026.04.25.10.12.57

looooo committed:
MNT: Re-rendered with conda-smithy 2026.5.29 and conda-forge-pinning 2026.05.30.09.17.10

looooo committed:
vtk9.6.0

looooo committed:
MNT: Re-rendered with conda-smithy 2026.5.29 and conda-forge-pinning 2026.05.30.09.17.10

h-vetinari commented:
Congratulations for getting this (back) to green! 🥳 

I suggest merging this and then doing individual migrations on top. :)
